### ISSUE TITLE
Riot: add the cookie warning + make our (anonymised) matomo stats opt in

### ISSUE THREAD

**Issue by lampholder** [vector-im/riot-meta#158]
GDPR doesn't mandate that anonymised stats are opt-in, but we're doing it anyway.

**Comment by ara4n** [comment]
So conclusion on design:

For Riot, we show a MatrixToolbar for all users who haven't viewed the cookie warning that says something like:

"Please enable analytics to help us improve Riot. This will set a cookie (see our [cookie]() and [privacy](policy)). \[Yes, I want to help!\] \[X\]"

We show this to all users who haven't explicitly consented to cookies one way or another (and also solve analytics opt-in).

Entirely separately, scalar-web should be displaying its own cookie warning within its iframe, which i'll add a new issue for.

**Comment by ara4n** [comment]
Could also link 'analytics' to the dialog we have somewhere which says what analytics gets reported back to piwik (and explains that piwik is self-hosted)

**Comment by ara4n** [comment]
For iOS & Android, we just need to ensure that the analytics is opt in. (No need for cookies for obvious reasons)

**Comment by manuroe** [comment]
iOS has already a consent modal for analytics compatible with GDPR.

**Comment by lampholder** [comment]
Yeah; let's pop up the same modal explanation we give if you click the 'how do we use analytics' link next to the analytics option in settings:

![image](https://user-images.githubusercontent.com/1922197/40349801-9d975688-5d9f-11e8-880e-2ff7a521687c.png)

**Comment by lampholder** [comment]
In terms of what we need to comply, the ICO says this:

**Comment by lampholder** [comment]
I link that because I'm not aware that we currently have a 'cookie policy' documented anywhere, and I don't think that we need one as long as we hit the bases outlined above (and I certainly don't want to spend time right now writing one if we don't have one already).

So I think perhaps the notif should read:

> We use fully anonymized analytics to help us improve Riot. This will set a cookie to track your browser session. [Yes, I want to help!] [X]

I _don't_ think we should link to or mention cookies in the privacy policy, because we don't need cookies for matrix.org synapse and we don't need a privacy policy for riot.im (because the analytics are anonymized), so let's keep those concerns good and separate.

**Comment by lampholder** [comment]
Temporary conclusion - we think we want to reword this a bit, and we think we _do_ want to host a cookie policy somewhere, but don't do any work on this until we have firm conclusions.

**Comment by lukebarnard1** [comment]
https://github.com/matrix-org/matrix-react-sdk/pull/1926 is landed, so "usage data" now opens the analytics modal when clicked on.
